docs: 데이터베이스 구조 변경

rladjswkd opened the pull request:
- [x] direct_message 테이블 제거
- [x] channel 테이블의 password 컬럼을 channel_password 테이블로 이동해 channel의 type이 protected일 때만 password를 데이터베이스에 저장하도록 수정
- [x] channel의 type에 one-to-one 타입 추가 ( 다이렉트 메시지로 활용 )

rladjswkd committed:
docs: 데이터베이스 구조 변경

- direct_message 테이블 제거
- channel 테이블의 password 컬럼을 channel_password 테이블로 이동해 channel의 type이 protected일 때만 password를 데이터베이스에 저장하도록 수정
- channel의 type에 one-to-one 타입 추가 ( 다이렉트 메시지로 활용 )

rladjswkd committed:
docs: 테이블 컬럼 이름 변경 및 추가

rladjswkd committed:
docs: 데이터베이스 구조 변경 및 ER Diagram 이미지 제거

- user가 PostgreSQL의 예약어이므로 user 테이블 이름을 account로 변경
- account 테이블에 nickname 컬럼 추가

Eingerjar commented on database.dbml:
```diff
@@ -6,9 +6,10 @@ Project project_name {
 
-Table user.user {
-  id integer [ pk, unique, default:0, increment ] // pk is not null.
+Table user.account {
```
예약어 존재로 인한 user => account 변경 확인

rladjswkd committed:
docs: 테이블 구조 변경 및 이름 변경

- friendship 테이블을 follow 테이블로 변경 및 first_user_id 컬럼과 second_user_id 컬럼의 이름을 following_account_id, followed_account_id로 변경
- match 테이블의 user_id, opponent_id 컬럼을 win_account_id, lose_account_id로 이름을 바꾸고 win_or_not 컬럼을 제거
- channel 테이블의 name 컬럼에 unique 제약 조건 추가

rladjswkd committed:
docs: match 테이블 구조 변경

- user_ladder_then, opponent_ladder_then 컬럼 제거
- type, map_type 컬럼 추가

rladjswkd commented:
- prisma dbml generator로 대체 -> 현재 브랜치 제거

Eingerjar reviewed (approved):
확인했습니다!ok :)
